rfcs/SchemaCoordinates.md

```diff
@@ -0,0 +1,372 @@
+# RFC: Schema Coordinates
+
+**Proposed by:** [Mark Larah](https://twitter.com/mark_larah) - Yelp
+
+This RFC proposes formalizing "Schema Coordinates" - a human readable syntax to
+uniquely identify a type, field, or field argument defined in a GraphQL Schema.
+
+This should be listed as a non-normative note in the GraphQL specification to
+serve as an official reference for use by third party tooling.
+
+## 📜 Problem Statement
+
+Third party GraphQL tooling and libraries may wish to refer to a field, or set of
+fields in a schema. Use cases include documentation, metrics and logging
+libraries.
+
+![](https://i.fluffy.cc/5Cz9cpwLVsH1FsSF9VPVLwXvwrGpNh7q.png)
+
+_(Example shown from GraphiQL's documentation search tab)_
+
+There already exists a convention used by some third party libraries for writing
+out fields in a unique way for such purposes. However, there is no formal
+specification or name for this convention.
+
+### Use cases
+
+1. A GraphQL server wants to **log how often each field in the schema is
+   requested**. This may be implemented by incrementing a counter by the name of
+   the schema coordinate for each field executed in a request.
+
+   _Existing implementations: Yelp (internal), Facebook (internal)_
+
+1. GraphiQL and other playgrounds / documentation sites want to show a list of
+   **search results** when a user searches for a type or field name. We can
+   display a list of schema coordinates that match the search term. A schema
+   coordinate can also be used in the hyperlink to form a permalink for
+   documentation for a particular field.
+
+   _Existing implementations: GraphiQL, Apollo Studio (see "Prior Art")_
+
+1. A developer may want to perform **analytics** on all known
+   [persisted queries][apq] - e.g. what are the most commonly used fields across
+   all documents. Schema coordinates may be used as the index/lookup keys when
+   storing this information in the database.
```

Comment:
In this use case, it may be useful to differentiate between `Query.forums` (all, unfiltered), `Query.forums(about:)` (filtered) and `Query.forums(about:,first:,after:)` (filtered, paginated). No change required, just noting because this adds weight to the desire to add more than one argument to a field, though I understand that this is not technically a "schema coordinate" at this point but instead a usage summary.

Reply:
++ something like `Query.forums(about:,first:,after:)` sounds like a great use case / future syntax extension.

Going to close this for now as it looks like there's no change needed, but let's certainly bear this in mind for the future!